Sources/AWSLambdaRuntimeCore/Lambda.swift

```diff
@@ -78,27 +54,19 @@ public enum Lambda {
         return String(cString: value)
     }
 
+    #if swift(>=5.5)
```

Comment:
should we just require 5.5? feels wrong to drop support for closures *and* support < 5.5

Reply:
We talked about this last time. We wanted to stay with `5.2` to allow existing frameworks (Vapor, Noze, Hummingbird, Smoke, and friends) to integrate with Lambda without needing to bump their requirements. For integration into those frameworks developers will always use the `Lambda.Lifecycle` APIs.

Normal Lambda users should require 5.5 and just use the async APIs.

I guess with this we will do what most of the Swift Server ecosystem is going todo... Make async/await the normal, leave escape hatch for advanced use cases using futures.